### PR TITLE
Move config.py from rtd build

### PR DESCRIPTION
**Pull request by stsewd**

### CODE REVIEW

**Comment by stsewd** [comment]
I think this is ready, next I think I can port the PRs from rtd-build (https://github.com/rtfd/readthedocs-build/pull/47 and https://github.com/rtfd/readthedocs-build/pull/38). Next https://github.com/rtfd/readthedocs-build/issues/50#issuecomment-398587620. @agjohnson what do you think?

**Comment by stsewd** [comment]
@agjohnson I was thinking about porting the rest of the PRs to v1, then work in the support for the two versions and v2 features. I'm not sure how much code will change to support the two versions p:. I will investigate some patterns and see what can fit here.

**Comment by stsewd** [comment]
> Perhaps for now we should just port the code, get it running in RTD core, then start working out this pattern, then start porting in some of the backwards incompatible changes.

Then this PR is done, I'm reading some patterns right now

**Comment by agjohnson** [comment]
We probably still need to address `formats` in this PR though, as that is going to break for existing yaml configurations with this PR. You could maybe remove the changes to `formats` and we can merge this.

As long as the features are backwards compatible on v1, it's not a problem to support them in v1 i think.

**Comment by stsewd** [comment]
mmm, do you mean this change https://github.com/rtfd/readthedocs-build/pull/43? I thought it was already deployed. We already change the docs for that https://docs.readthedocs.io/en/latest/yaml-config.html#formats. Also, the value was never used in the rtd code. But it wasn't causing any error in both sides anyway. Should I keep the compatibility with `formats: none`?

**Comment by stsewd** [comment]
I think I understand the issue, I had added some tests and keep the current behavior for the `formats` key.

File: readthedocs/config/config.py [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+        try:
+            return self.env_config['python']['supported_versions']
+        except (KeyError, TypeError):
+            pass
+        return self.PYTHON_SUPPORTED_VERSIONS
+
+    def get_valid_formats(self):  # noqa
+        """Get all valid documentation formats."""
+        return (
+            'htmlzip',
```
Comment (agjohnson) [inline readthedocs/config/config.py]: it looks like we lost `none` in this list, do we need it?

Comment (stsewd) [reply readthedocs/config/config.py]: I moved this logic, it doesn't make much sense having this as a valid option, but in the past, we allowed `['none']` as a way of ignoring all the types, this check is now in https://github.com/rtfd/readthedocs.org/pull/4272/files#diff-2b6854693f39acfa183fca7fef27dcfbR423. BTW, we never put `none` as a valid option in the past docs.

**Comment by agjohnson** [comment]
Is this ready to merge then? It's marked as WIP.

**Comment by stsewd** [comment]
Yeah, we can merge this now.

**Pull request merged**